packages/platform/types/event-types/event-types_2024_06_14/inputs/create-event-type.input.ts

```diff
@@ -512,6 +514,66 @@ export class BaseCreateEventTypeInput {
       "Boolean to require authentication for booking this event type via api. If true, only authenticated users who are the event-type owner or org/team admin/owner can book this event type.",
   })
   bookingRequiresAuthentication?: boolean;
+
+  @IsOptional()
+  @IsBoolean()
+  @DocsPropertyOptional({
+    description: "If true, guests and organizer can no longer cancel the event.",
+  })
+  disableCancelling?: boolean;
+
+  @IsOptional()
+  @IsBoolean()
+  @DocsPropertyOptional({
+    description: "If true, guests and organizer can no longer reschedule the event.",
+  })
+  disableRescheduling?: boolean;
```

Comment:
This PR https://github.com/calcom/cal.com/pull/25575 adds a new ability to not only toggle this setting on / off but also specify how many minutes before meeting we disable rescheduling.

So, to future proof this let's have `disableRescheduling` as object of type `DisableRescheduling` which right now only has 1 property to enable or disable this setting and then later on we can also add "when less than X minutes before the meeting setting".

I would recommend the same with "disableCancelling" just in case we add "when less than X minutes before the meeting setting" also there and because they are similar settings.

What do you think?